modules/currency-l1/src/main/scala/org/tessellation/currency/l1/domain/dataApplication/consensus/Validator.scala

```diff
@@ -0,0 +1,86 @@
+package org.tessellation.currency.l1.domain.dataApplication.consensus
+
+import cats.effect.Async
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.{Applicative, Monad}
+
+import org.tessellation.currency.dataApplication.{BaseDataApplicationL1Service, ConsensusInput}
+import org.tessellation.node.shared.domain.cluster.storage.ClusterStorage
+import org.tessellation.node.shared.domain.node.NodeStorage
+import org.tessellation.node.shared.domain.snapshot.storage.LastSnapshotStorage
+import org.tessellation.schema.node.NodeState
+import org.tessellation.schema.node.NodeState.Ready
+import org.tessellation.schema.peer.PeerId
+import org.tessellation.schema.{GlobalIncrementalSnapshot, GlobalSnapshotInfo}
+import org.tessellation.security.signature.Signed
+import org.tessellation.security.{Hasher, SecurityProvider}
+
+import eu.timepit.refined.types.numeric.PosInt
+import io.circe.Encoder
+import io.circe.generic.semiauto.deriveEncoder
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+object Validator {
+
+  private def logger[F[_]: Async] = Slf4jLogger.getLogger
+
+  private def isReadyForDataApplicationConsensus(state: NodeState): Boolean = state == Ready
```

Comment:
if you don't share the code, it can be just `isReadyForConsensus` :) you don't need to repeat data application since it's in the dataApplication/consensus package